datastore/importers/nmdb-import-systeminfo/Parser.test.cpp

```diff
@@ -0,0 +1,276 @@
+      )STR", //Mising first string
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name:
+      DHCP Enabled:    Yes
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing connection name
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing dhcp enabled data
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing ipaddresses label
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      IP address(es)
+      )STR" //Missing Ipaddress data
+    };
+    for (const auto& test : testsOk) {
+      BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'networkCard': " << test);
+    }
+    for (const auto& test: testsFail) {
+      BOOST_TEST(!nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'networkCard fails': " << test);
+    }
+    tp.data.network_cards.clear();
+  }
+  { // Test Data
+    const auto& parserRule {tp.networkCard};
+    std::string out;
+    std::string test {
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR",
+    };
+    BOOST_TEST(nmdp::testAttr(test.c_str(), parserRule, out, blank),
+              "Parse rule 'networkCard': " << test);
+    const auto debugString { tp.data.network_cards["realtek pcie gbe family controller"].toDebugString()};
+    BOOST_TEST(tp.data.network_cards["realtek pcie gbe family controller"].isValid());
+    for (const auto& ipAddr : tp.data.network_cards["realtek pcie gbe family controller"].getIpAddresses()) {
+      BOOST_TEST(ipAddr.isValid());
+    }
+    nmdp::testInString(debugString, "name: realtek pcie gbe family controller");
+    nmdp::testInString(debugString, "mediaType: ethernet");
+    nmdp::testInString(debugString, "ipAddress: 192.168.0.2/32");
+    nmdp::testInString(debugString, "ipAddress: 2001:db8:85a3::8a2e:370:7334/128");
+  }
+}
+
+BOOST_AUTO_TEST_CASE(testhotfixes)
+{
+  TestParser tp;
+  {
+    const auto& parserRule {tp.hotfixes};
+    std::string test {
+      R"STR(Hotfix(s):           10 Hotfix(s) Installed.
+      [01]: KB000001
+      [02]: KB000002
+      [03]: KB000003
+      [04]: KB000003
+      [05]: KB000004
+      [06]: KB000005
+      [07]: KB000006
+      [08]: KB000007
+      [09]: KB000008
+      [10]: KB000009
+      )STR"
+
+    };
+
+    BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank, false),
```

Comment:
This test should fully consume the input.
```suggestion
    BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
```